frontend/src/ui/Input.tsx

```diff
@@ -287,3 +288,36 @@ export const CopyableInput: React.FC<CopyableInputProps> = ({
         </div>
     );
 };
+
+type DisplayOptionGroup<TFieldValues extends FieldValues> = {
+    form: UseFormReturn<TFieldValues>;
+    type: React.HTMLInputTypeAttribute;
+    optionProps: [
+        option: Path<TFieldValues>,
+        title: string,
+        checked?: boolean,
+        value?: string,
+    ][];
```

Comment:
This syntax is completely new to me :D But why not make this an array of an object, where the fields are actually named? I suppose that would make the code creating the `optionProps` longer because you have to specify the field names ... mh. But it would make it more readable as one doesn't have to guess what the booleans and stuff mean.

Reply:
While I don't love the verbosity this brings, you do have a very valid point with readability. I agree, that is more important than terseness.